sdcm/utils/common.py

```diff
@@ -2247,3 +2248,21 @@ def walk_thru_data(data, path: str) -> Any:
             continue
         current = current.get(name, None)
     return current
+
+
+def update_authenticator(nodes, authenticator='AllowAllAuthenticator', restart=True):
+    """
+    Update the authenticator of nodes, restart the nodes to make the change effective
+    """
+    for node in nodes:
+        node.remoter.sudo(
+            f"sed -ie 's/^authenticator:.*/authenticator: {authenticator}/g' /etc/scylla/scylla.yaml")
```

Comment:
don't `sed`, use the remoter function to open a remote file (it will need to use `sed` for branch-2020.1, but from here (master) on, it will use the most advanced features we have in place

Reply:
I tried the _remote_yaml(), the file can be copied to local, but not update to remote in the end.

```
ubuntu@ip-10-0-1-26:~/sct-results$ ls -l /tmp/sctaet56tbo/saslauthd.conf
-rw-r--r-- 1 ubuntu root 0 Apr  1 07:40 /tmp/sctaet56tbo/saslauthd.conf


            with node._remote_yaml('/etc/saslauthd.conf') as saslauthd_yml:
                saslauthd_yml.update(node.get_saslauthd_config())


 t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > Exception in setUp. Will call tearDown < t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > Exception in setUp. Will call tearDown
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > Traceback (most recent call last):
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 124, in wrapper
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     return method(*args, **kwargs)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 110, in inner
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     res = func(*args, **kwargs)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 489, in setUp
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     self.init_nodes(db_cluster=self.db_cluster)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 439, in init_nodes
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     db_cluster.wait_for_init(node_list=db_cluster.nodes)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3478, in wrapper
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     verify_node_setup(start_time)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3452, in verify_node_setup
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     raise NodeSetupFailed(node=node, error_msg=setup_exception[0], traceback_str=setup_exception[1])
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > sdcm.cluster.NodeSetupFailed: [Node longevity-10gb-3h-master-db-node-cd3c63e5-1 [13.48.135.169 | 10.0.0.93] (seed: True)] NodeSetupFailed: 'NoneType' object has no attribute 'update'
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > Traceback (most recent call last):
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3441, in node_setup
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     cl_inst.node_setup(_node, **setup_kwargs)
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 4062, in node_setup
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR >     saslauthd_yml.update(node.get_saslauthd_config())
< t:2021-04-01 07:40:50,061 f:tester.py       l:131  c:sdcm.tester          p:ERROR > AttributeError: 'NoneType' object has no attribute 'update'

```

Reply:
I will solve the problem and switch to use remote file.

Reply:
the branch-2020.1 doesn't have this feature, but both enterprise and branch-2021.1 do have this feature...